### PR TITLE
Plugins UI revamp

### LINKED ISSUE #412: Plugin upgrades interface is in the way
This view is hard to follow when there are many plugins that need to be updated, yet you also have many you're actively working on:

image

I propose to collapse the "plugins to update" box by default, and show it as an [antd alert](https://ant.design/components/alert/) instead, something like:

image

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Changes

A significant revamp of the plugins UI. Highlights:

- Addresses https://github.com/PostHog/plugin-server/issues/413
- Fixes https://github.com/PostHog/plugin-server/issues/412
- Addresses https://github.com/PostHog/plugin-server/issues/411
- Toggleable sections: hide and show plugin sections
- Search for both the 'Repository' and 'Installed' tabs
- Plugins that don't process in order are now flagged and not sortable (frontend change only, which will work, but is suboptimal)
- `InstalledTab` refactored into a file per section 
- Takes plugins out of Beta 🎉 

Screenshot 2021-07-14 at 18 49 53
Screenshot 2021-07-14 at 18 50 46
Screenshot 2021-07-14 at 18 51 26
Screenshot 2021-07-14 at 18 51 43

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

File: frontend/src/scenes/plugins/PluginsSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { Input } from 'antd'
+
+export function PluginsSearch(): JSX.Element {
+    const { searchTerm, rearranging } = useValues(pluginsLogic)
+    const { setSearchTerm } = useActions(pluginsLogic)
+    return (
+        <Input
+            placeholder="Start typing to search for a plugin"
+            onChange={(e) => setSearchTerm(e.target.value)}
+            style={{ maxWidth: 400, height: 40, borderRadius: 5, marginTop: 5, marginBottom: 10 }}
```
Comment (Twixes) [inline frontend/src/scenes/plugins/PluginsSearch.tsx:13]: This seems kinda bulky to me. See person search:

vs. plugin search

File: frontend/src/scenes/plugins/edit/CapabilitiesInfo.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export const capabilitiesInfo: Record<string, string> = {
+    processEvent: 'Used to read and modify events before ingestion',
+    processEventBatch: 'Used to read and modify and modify events in batches before ingestion',
```
Comment (Twixes) [inline frontend/src/scenes/plugins/edit/CapabilitiesInfo.ts:3]: Maybe "Provides" here too instead of "Used to"? I dunno why, but "Used to" doesn't fit this IMO, it sounds to me somewhat… passive?

File: frontend/src/scenes/plugins/edit/PluginDrawer.tsx [hunk 3]
```diff
@@ -97,6 +98,7 @@ export function PluginDrawer(): JSX.Element {
         }
         setInvisibleFields(fieldsToSetAsInvisible)
     }
+    console.log(editingPlugin?.capabilities)
```
Comment (Twixes) [inline frontend/src/scenes/plugins/edit/PluginDrawer.tsx:101]: Debug

File: frontend/src/scenes/plugins/tabs/installed/sections/UpgradeSection.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                        <>
+                            {sectionsOpen.includes(PluginSection.Upgrade) ? (
+                                <CaretDownOutlined />
+                            ) : (
+                                <CaretRightOutlined />
+                            )}
+                            {` Plugins to update (${filteredPluginsNeedingUpdates.length})`}
+                        </>
+                    }
+                    buttons={!rearranging && sectionsOpen.includes(PluginSection.Upgrade) && upgradeButton}
```
Comment (Twixes) [inline frontend/src/scenes/plugins/tabs/installed/sections/UpgradeSection.tsx:63]: Do we have a mechanism that checks for plugin updates without user interaction?

**Review by Twixes — COMMENTED** [review]
Cool stuff! Left a few comments.
BTW, in the PR's description, should "Addresses" be "Resolves"?

File: frontend/src/scenes/plugins/pluginsLogic.ts [hunk 5]
```diff
@@ -528,6 +573,57 @@ export const pluginsLogic = kea<pluginsLogicType<PluginForm>>({
...
+                    : pluginsNeedingUpdates,
+        ],
+        sortableEnabledPlugins: [
+            (s) => [s.filteredEnabledPlugins],
+            (filteredEnabledPlugins) => {
+                return filteredEnabledPlugins.filter(
+                    (plugin) =>
+                        !plugin.capabilities ||
+                        plugin.capabilities.methods.includes('processEvent') ||
+                        plugin.capabilities.methods.includes('processEventBatch')
```
Comment (Twixes) [inline frontend/src/scenes/plugins/pluginsLogic.ts:615]: This crashed the Plugins page for me when I tried to enable a code editor plugin, because its `capabilities` object didn't have `methods`.

File: frontend/src/scenes/plugins/tabs/installed/sections/EnabledPluginsSection.tsx [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+                    e.stopPropagation()
+                    savePluginOrders(temporaryOrder)
+                }}
+                disabled={!pluginOrderSaveable}
+            >
+                Save order
+            </Button>
+            <Button type="default" icon={<CloseOutlined />} onClick={cancelRearranging}>
+                Cancel
+            </Button>
```
Comment (Twixes) [inline frontend/src/scenes/plugins/tabs/installed/sections/EnabledPluginsSection.tsx:97]: Kinda weird that clicking this causes the "Enabled plugins" section to close.

Comment (yakkomajuri) [reply frontend/src/scenes/plugins/tabs/installed/sections/EnabledPluginsSection.tsx:97]: Ah yes need to stop event propagation here probably

File: frontend/src/scenes/plugins/pluginsLogic.ts (hunk 5 above)
Comment (yakkomajuri) [reply frontend/src/scenes/plugins/pluginsLogic.ts:615]: Good catch!

File: frontend/src/scenes/plugins/tabs/installed/sections/UpgradeSection.tsx (hunk 4 above)
Comment (yakkomajuri) [reply frontend/src/scenes/plugins/tabs/installed/sections/UpgradeSection.tsx:63]: We don't - could do that too

File: frontend/src/scenes/plugins/edit/CapabilitiesInfo.ts (hunk 2 above)
Comment (yakkomajuri) [reply frontend/src/scenes/plugins/edit/CapabilitiesInfo.ts:3]: I played around with the wording quite a bit here - never got to something I was super happy with. Will try again.

File: frontend/src/scenes/plugins/PluginsSearch.tsx (hunk 1 above)
Comment (yakkomajuri) [reply frontend/src/scenes/plugins/PluginsSearch.tsx:13]: Personal preference - I like bulky (see https://posthog.com/contributors)

Will update for consistency though

File: frontend/src/scenes/plugins/tabs/installed/sections/UpgradeSection.tsx (hunk 4 above)
Comment (Twixes) [reply frontend/src/scenes/plugins/tabs/installed/sections/UpgradeSection.tsx:63]: I thought @mariusandra added something like that some time ago, but not sure

Comment (yakkomajuri) [reply frontend/src/scenes/plugins/tabs/installed/sections/UpgradeSection.tsx:63]: Oh I think we check once on load, but not continuously after is what I mean

Comment (Twixes) [reply frontend/src/scenes/plugins/tabs/installed/sections/UpgradeSection.tsx:63]: Ah, cool, that's good. :+1: I don't think we need to check continuously. But if the "Check for updates" button is hidden when the "Plugins to update" sections is closed (which will be true most of the time – a "All your plugins are up to date. Great work!" message is not interesting), then it's going to be a seldom used, very intentional interaction. So it's good to check every now and then _without_ user interaction.

Comment (yakkomajuri) [reply frontend/src/scenes/plugins/tabs/installed/sections/UpgradeSection.tsx:63]: I see yeah

**Comment by yakkomajuri** [comment]
How are all backend tests failing after this commit? 

https://github.com/PostHog/posthog/pull/5137/commits/3eabefbe93a08c9659f778e208876bc6fec84f35

**Comment by Twixes** [comment]
Looks like some larger issue with setting up Python (caching?), same with #5146.

**Comment by yakkomajuri** [comment]
Seems to happen on some PRs but not others?

**Review by Twixes — APPROVED** [review]
Looks good to me, besides CI refusing to install Python…
